File: expandable/src/main/java/ir/drax/expandable/ResizeAnimation.java
package ir.drax.expandable;

import android.view.View;
import android.view.animation.AccelerateDecelerateInterpolator;
import android.view.animation.Animation;
import android.view.animation.Transformation;

public class ResizeAnimation extends Animation {
    final int targetHeight;
    View view;
    int startHeight;
    private Interpolate interpolate;

    public ResizeAnimation(View view, int targetHeight, int startHeight,Interpolate interpolate) {
        this.view = view;
        this.targetHeight = targetHeight;
        this.startHeight = startHeight;
        this.interpolate = interpolate;
        setInterpolator(new AccelerateDecelerateInterpolator());
    }

    @Override
    protected void applyTransformation(float interpolatedTime, Transformation t) {
        int newHeight = (int) (startHeight+(targetHeight - startHeight) * interpolatedTime);
        //to support decent animation, change new heigt as Nico S. recommended in comments
        //int newHeight = (int) (startHeight+(targetHeight - startHeight) * interpolatedTime);
        interpolate.interpolate(interpolatedTime);
        view.getLayoutParams().height = newHeight;
        view.requestLayout();
    }

    @Override
    public void initialize(int width, int height, int parentWidth, int parentHeight) {
        super.initialize(width, height, parentWidth, parentHeight);
    }

    @Override
    public boolean willChangeBounds() {
        return true;
    }
}

File: expandable/src/main/java/ir/drax/expandable/ResizeAnimation.kt
package ir.drax.expandable

import android.util.Log
import android.view.View
import android.view.animation.AccelerateDecelerateInterpolator
import android.view.animation.Animation
import android.view.animation.Transformation

class ResizeAnimation(private val view:View, private val targetHeight:Int, private val startHeight:Int, private val interpolate:Interpolate?=null): Animation() {
    init {
        interpolator=AccelerateDecelerateInterpolator()
    }

    override fun willChangeBounds(): Boolean {
        return true
    }

    override fun applyTransformation(interpolatedTime: Float, t: Transformation?) {
        if (interpolatedTime==1f)return // To stop any extra layout changes right after the animation ended-> To avoid threshold bugs
        Log.e("Transform","sss$interpolatedTime")
        val newHeight = startHeight + (targetHeight - startHeight) * interpolatedTime
        //to support decent animation, change new heigt as Nico S. recommended in comments
        //int newHeight = (int) (startHeight+(targetHeight - startHeight) * interpolatedTime);
        interpolate?.interpolate(interpolatedTime)
        view.layoutParams.height = newHeight.toInt()
        view.requestLayout()
    }
}